grafana-plugin/src/containers/Rotations/Rotations.tsx

```diff
@@ -104,35 +114,55 @@ class Rotations extends Component<RotationsProps, RotationsState> {
-                    <div>
+              <HorizontalGroup>
+                <Button
+                  variant="secondary"
+                  onClick={() => {
+                    const closestEvent = findClosestUserEvent(dayjs(), currentUserPk, layers);
+                    const swapStart = closestEvent
+                      ? dayjs(closestEvent.start)
+                      : dayjs().tz(currentTimezone).startOf('day').add(1, 'day');
+
+                    const swapEnd = closestEvent ? dayjs(closestEvent.end) : swapStart.add(1, 'day');
+
+                    onShowShiftSwapForm('new', {
+                      swap_start: getUTCString(swapStart),
+                      swap_end: getUTCString(swapEnd),
+                    });
+                  }}
+                >
+                  Request shift swap
+                </Button>
+                {disabled ? (
+                  isTypeReadOnly ? (
+                    <Tooltip content="Ical and API/Terraform rotations are read-only here" placement="top">
+                      <div>
+                        <Button variant="primary" icon="plus" disabled>
+                          Add rotation
+                        </Button>
+                      </div>
+                    </Tooltip>
+                  ) : (
+                    <WithPermissionControlTooltip userAction={UserActions.SchedulesWrite}>
                       <Button variant="primary" icon="plus" disabled>
                         Add rotation
                       </Button>
-                    </div>
-                  </Tooltip>
+                    </WithPermissionControlTooltip>
+                  )
+                ) : options.length > 0 ? (
+                  <ValuePicker
+                    label="Add rotation"
+                    options={options}
+                    onChange={this.handleAddRotation}
+                    variant="primary"
+                    size="md"
+                  />
                 ) : (
-                  <WithPermissionControlTooltip userAction={UserActions.SchedulesWrite}>
-                    <Button variant="primary" icon="plus" disabled>
-                      Add rotation
-                    </Button>
-                  </WithPermissionControlTooltip>
-                )
-              ) : options.length > 0 ? (
-                <ValuePicker
-                  label="Add rotation"
-                  options={options}
-                  onChange={this.handleAddRotation}
-                  variant="primary"
-                  size="md"
-                />
-              ) : (
-                <Button variant="primary" icon="plus" onClick={() => this.handleAddLayer(nextPriority, startMoment)}>
-                  Add rotation
-                </Button>
-              )}
+                  <Button variant="primary" icon="plus" onClick={() => this.handleAddLayer(nextPriority, startMoment)}>
```

Comment:
seems the event handler usages are not all consistent, in some parts there are direct function calls, while in some places there's arrow function declarations. might be worth sticking to just arrow functions as they're more practical?